lib/phoenix/endpoint.ex

```diff
@@ -778,24 +778,30 @@ defmodule Phoenix.Endpoint do
     paths =
       if websocket do
         triplet = {:websocket, socket, socket_config(websocket, Phoenix.Transports.WebSocket)}
-        [{socket_path(path, :websocket), triplet} | paths]
+        [{socket_path(path, "websocket", websocket), triplet} | paths]
       else
         paths
       end
 
     paths =
       if longpoll do
         plug_init = {endpoint, socket, socket_config(longpoll, Phoenix.Transports.LongPoll)}
-        [{socket_path(path, :longpoll), {:plug, Phoenix.Transports.LongPoll, plug_init}} | paths]
+        [{socket_path(path, "longpoll", longpoll), {:plug, Phoenix.Transports.LongPoll, plug_init}} | paths]
```

Comment:
Instead of passing `longpoll` as third argument, I would store `socket_config(longpoll, Phoenix.Transports.LongPoll)` in a variable and pass it as argument instead. This way we don't need to handle non-lists inside `socket_path`. Or, even better, I would put `[path: "longpoll"]` as a default inside `Phoenix.Transports.LongPoll.default_config`, this way you don't need to provide a default here too and you can just do `Keyword.fetch!(config, :path)` inside `socket_path`.  Thoughts?